### PR TITLE
Changing to Github actions workflow instead of Travis ci

### PR DESCRIPTION
**Pull request by FredDodd6**
Removed old Travis files. 
Added workflow to the MANIFEST file.
Set up GitHub workflow to test and build the repo.

### CODE REVIEW

**Comment by mgcam** [comment]
Add scripts/install_npg_perl_dependencies.sh to MANIFEST?

File: .github/workflows/testing_and_building_repo.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          eval $(perl -I ~/perl5ext/lib/perl5/ -Mlocal::lib=~/perl5ext)
+          cpanm --installdeps --notest .
+
+    # running ./Build dist and exporting files
+    - name: run ./Build dist  
+      run: |
+          eval $(perl -I ~/perl5ext/lib/perl5/ -Mlocal::lib=~/perl5ext)
+          export TEST_AUTHOR=1
+          ./Build dist
+          export DIST_FILE=$(ls ml_warehouse-*.tar.gz)
```
Comment (kjsanger) [inline .github/workflows/testing_and_building_repo.yml:46]: You don't need a sub-shell and `ls` for the glob:

```
$ touch foo_1234.txt
$ x=foo_*.txt
$ echo "$x"
foo_1234.txt

Comment (FredDodd6) [reply .github/workflows/testing_and_building_repo.yml:46]: So no `ls`, just :
`export DIST_FILE=$(ml_warehouse-*.tar.gz)` ?

Comment (kjsanger) [reply .github/workflows/testing_and_building_repo.yml:46]: Not even that, see the example above. The subshell isn't needed.
`export x=foo_*.txt`

(There is a case for using a subshell if you use e.g. `shopt -s nullglob` to trigger a failure if the glob doesn't match anything and also want to confine the effect of the `shopt` to just the shell where the glob is done)

**Pull request merged**